File: CPP /Unit 06/Lec 39/Q5.cpp
Single File Programming Question
Problem Statement



Akshaya is developing a program to calculate the sum of elements in arrays of different data types. Create a program that calculates the sum of elements in two different arrays of integers and double-point numbers using a template function named arraySum. The program should first prompt the user to enter the size of the arrays, followed by the elements of the two arrays.



The arraySum function should take two parameters: the array of any data type and the size of the array, and it should return the sum of all the elements in the array.



The program should print the sum of elements in the integer array and the double-point array separately.

Input format :
The first line consists of an integer n, representing the size of the array.

The second line consists of n space-separated integers, representing the array of integer values.

The third line consists of n space-separated double values, representing the array of double values.

Output format :
The first line prints an integer, representing the sum of an integer array.

The second line prints a double value, representing the sum of the double-point array.



Refer to the sample output for the formatting specifications.

Code constraints :
1 <= n <= 1000: The size of the arrays will be between 1 and 1000 (inclusive).

For the integer array arr1, -109 <= arr1[i] <= 109

For the floating-point array arr2, -109 <= arr2[i] <= 109

Sample test cases :
Input 1 :
5
1 2 3 4 5
1.5 2.5 3.5 4.5 5.5
Output 1 :
15
17.5
Input 2 :
3
10 20 30
1.1 2.2 3.3
Output 2 :
60
6.6



#include <iostream>
using namespace std;

template <class T>
T arraySum(T arr[], int n) {
    T sum = 0;
    for (int i = 0; i < n; i++) {
        sum += arr[i];
    }
    return sum;
}

int main() {
    int n;
    cin >> n;

    int arr1[n];
    double arr2[n];

    for (int i = 0; i < n; i++) {
        cin >> arr1[i];
    }

    for (int i = 0; i < n; i++) {
        cin >> arr2[i];
    }

    cout << arraySum(arr1, n) << endl;
    cout << arraySum(arr2, n);

    return 0;
}

File: Java/Unit 03/Lec 19/Q5.java
Single File Programming Question
Problem Statement



Romit wants to implement a program to find the smallest element in an array of integers using an interface with a default method. 



Help him design an interface named ArrayProcessor that includes a default method findSmallestElement(), which takes an array of integers as input and returns the smallest element. 



The Main class should implement this interface and use the default method to determine the smallest element in the array.

Input format :
The first line of input consists of an integer N, representing the number of elements in the array.

The second line of input consists of N space-separated integers, representing the array elements.

Output format :
The output prints an integer representing the smallest element in the array.



Refer to the sample output for the formatting specifications.

Code constraints :
The given test cases will fall under the following constraints:

2 ≤ N ≤ 10

1 ≤ array elements ≤ 100

Sample test cases :
Input 1 :
5
45 23 65 98 1
Output 1 :
1
Input 2 :
10
14 27 38 49 56 65 72 83 98 100
Output 2 :
14
Input 3 :
2
10 40
Output 3 :
10
Whitelist
Set 1:
default
interface


import java.util.Scanner;

// Interface ArrayProcessor with a default method
interface ArrayProcessor {
    // Default method to find the smallest element in an array
    default int findSmallestElement(int[] arr) {
        if (arr == null || arr.length == 0) {
            return 0; // Edge case handling
        }
        
        // Pehle element ko smallest maan kar shuru karte hain
        int smallest = arr[0];
        
        // Baaki elements se compare karte hain
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < smallest) {
                smallest = arr[i];
            }
        }
        return smallest;
    }
}

// Footer snippet ke mutabiq Main class interface ko implement karegi
public class Main implements ArrayProcessor {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        // N (number of elements) read karna
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        // Main ka object banakar default method call karna
        Main obj = new Main();
        int result = obj.findSmallestElement(arr);

        // Result print karna
        System.out.println(result);
        
        sc.close();
    }
}
